xml/spack.xml

```diff
@@ -0,0 +1,512 @@
+    argobots  hwloc  libfabric  libpciaccess  libxml2  pmix  slurm  ucx
+
+Run Dependencies:
+    None
+
+Virtual Packages:
+    mpich@3: provides mpi@:3.0
+    mpich@1: provides mpi@:1.3
+    mpich provides mpi</screen>
+
+    <para>
+     Lets build <literal>mpich</literal> with some variants:
+    </para>
+
+    <screen>
+&prompt;spack install mpich@3.3.2 -romio -libxml2 -hydra -fortran
+==> mpich: Executing phase: 'autoreconf'
+==> mpich: Executing phase: 'configure'
+==> mpich: Executing phase: 'build'
+==> mpich: Executing phase: 'install'
+[+] /usr/opt/spack/linux-sle_hpc15-skylake/gcc-7.5.0/mpich-3.3.2-3nexqqx2r4m7p2y7lmovuwvvobi2bygw</screen>
+
+<para>
+ Lets do it again with default parameter:
+</para>
+
+<screen>
+&prompt;spack install mpich@3.3.2
+.....
+[+] /usr/opt/spack/linux-sle_hpc15-skylake/gcc-7.5.0/mpich-3.3.2-sahjm2uhsoc3bi2cljtypwuqaflhamnx</screen>
+
+<para>
+ Another build of <literal>mpich-3.3.2</literal> with a <literal>cppflags</literal> and a specific <literal>libxml</literal> version (2.9.4)
+</para>
+
+<screen>
+&prompt;spack install mpich@3.3.2 cppflags="-O3 -fPIC" target=x86_64 ^libxml2@2.9.4</screen>
+
+<para>
+ Lets list all available packages:
+</para>
+
+<screen>
+&prompt;spack find -l
+==> 33 installed packages
+-- linux-sle_hpc15-skylake / gcc@7.5.0 --------------------------
+3griwie doxygen@1.8.20  y3w7dlk libpciaccess@0.16  vry3tfp netcdf-c@4.7.4      x3glm5y xz@5.2.3
+3cyidn4 hdf5@1.10.7     to2atk6 libpciaccess@0.16  msiysdr netcdf-cxx4@4.3.1   bzeebyp xz@5.2.3
+rjdqcht hwloc@1.11.11   4cyoxau libxml2@2.9.4      tiyqyxb netcdf-cxx4@4.3.1   z6y74kg zlib@1.2.11
+4lxxw65 hwloc@2.2.0     d4c7csk libxml2@2.9.10     tcuvjjt numactl@2.0.14      m4zmub6 zlib@1.2.11
+yof3lps hwloc@2.2.0     3nexqqx mpich@3.3.2        ks5elgg openmpi@3.1.6
+itovpc5 libiconv@1.16   sahjm2u mpich@3.3.2        rpnlbst util-macros@1.19.1
+5mvdlgl libiconv@1.16   jxbspwk mpich@3.3.2        boclx6a util-macros@1.19.1
+
+-- linux-sle_hpc15-x86_64 / gcc@7.5.0 ---------------------------
+yzxibyd hwloc@2.2.0    7wmqik2 libpciaccess@0.16  6xjiutf mpich@3.3.2         zmfrpzf xz@5.2.3
+mm7at5h libiconv@1.16  jrtvvdj libxml2@2.9.4      jjxbukq util-macros@1.19.1  jxqtsne zlib@1.2.11</screen>
+
+<para>
+ To show output dependencies along with found specs of <literal>/6xjiutf</literal> (<literal>mpich</literal>):
+</para>
+
+<screen>
+&prompt;spack find --deps /6xjiutf
+==> 1 installed package
+-- linux-sle_hpc15-x86_64 / gcc@7.5.0 ---------------------------
+mpich@3.3.2
+    hwloc@2.2.0
+        libpciaccess@0.16
+        libxml2@2.9.4
+            libiconv@1.16
+            xz@5.2.3
+            zlib@1.2.11</screen>
+
+     </sect1>
+     <sect1 xml:id="spack-compiler">
+      <title>Use a specific compiler</title>
+      <para>
+       The goal is to build <literal>mpich</literal> with <literal>gcc-10.2.0</literal>.
+       First install <literal>gcc-10.2.0</literal>:
+      </para>
+      <screen>
+&prompt;spack install gcc@10.2.0
```

Comment:
I didn't get to test this section yet because this command got stuck at `Executing phase: 'build'` (could just be my test server being slow). I'll have another go on Monday but it's probably fine.